src/main/java/net/consensys/orion/enclave/EncryptedPayload.java

```diff
@@ -90,19 +92,22 @@ public EncryptedPayload(
     return privacyGroupId;
   }
 
-  public EncryptedPayload stripFor(final Box.PublicKey key) {
-    final Integer toKeepIdx = encryptedKeyOwners.get(key);
+  public EncryptedPayload stripFor(final List<Box.PublicKey> keys) {
+    final List<EncryptedKey> keepKeys = keys
+        .stream()
+        .filter(key -> encryptedKeys[encryptedKeyOwners.get(key)] != null)
```

Comment:
If you do the map first then you can just filter using Objects::nonNull i.e. .map(key -> encryptedKeys[encryptedKeyOwners.get(key)]).filter(Objects::nonNull).collect(Collectors.toList())

Reply:
cool! Done